internal/component/otelcol/internal/scheduler/scheduler.go

```diff
@@ -53,19 +73,45 @@ func New(l log.Logger) *Scheduler {
 // Schedule completely overrides the set of previously running components;
 // components which have been removed since the last call to Schedule will be
 // stopped.
-func (cs *Scheduler) Schedule(ctx context.Context, h otelcomponent.Host, cc ...otelcomponent.Component) {
+func (cs *Scheduler) Schedule(ctx context.Context, updateConsumers func(), h otelcomponent.Host, cc ...otelcomponent.Component) {
```

Comment:
somewhere in this file you could add the process order that you told me:

- Pause consumers
- Stop the process
- Change the consumers
- Start the process
- Start the consumer

This is a good way to understand what's happening and to make sure that there is no race condition that's breaking the logic